tools/olm-bundle/generate-olm-bundle.sh

```diff
@@ -0,0 +1,230 @@
+  yq ea -i "select(fi==0).spec.install.spec.deployments[0].spec = select(fi==1).spec | select(fi==0)" "${CSV_FILE}" "${CRD_DIR}/060-Deployment-strimzi-cluster-operator.yaml"
+  
+  # Remove resource sections as it was causing problems with OLM in the past
+  yq ea -i 'del(.spec.install.spec.deployments[0].spec.template.spec.containers[0].resources)' "${CSV_FILE}"
+  
+  yq ea -i 'del(.spec.install.spec.deployments[0].spec.template.spec.containers[0].env[] | select(.name == "STRIMZI_OPERATOR_NAMESPACE"))' "${CSV_FILE}"
+  
+  yq ea -i ".spec.replaces = \"${PREVIOUS_BUNDLE_VERSION}\" | .spec.replaces style=\"\"" "${CSV_FILE}"
+  
+  generate_related_images
+  generate_image_digests
+}
+
+validate_olm_bundle() {
+  operator-sdk bundle validate "${BUNDLE}" --select-optional name=operatorhub
+}
+
+generate_related_images() {
+  # Create relatedImages section
+  yq ea -i '.spec.relatedImages = null' "${CSV_FILE}"
+
+  # Add operator image
+  image=$(yq ea ".. | select(has(\"image\")) | (select (.name == \"strimzi-cluster-operator\")).image" "${CSV_FILE}")
+  yq ea -i ".spec.relatedImages += [{\"name\": \"strimzi-operator\", \"image\": \"$image\"}]" "${CSV_FILE}";
+  yq ea -i ".metadata.annotations.containerImage = \"$image\"" "${CSV_FILE}"
+
+  # Add Kafka images
+  KAFKA_IMAGE_VALUES=$(yq eval '.. | select(has("name")).env[] | (select (.name == "STRIMZI_KAFKA_IMAGES")).value' "${CSV_FILE}")
+  for val in $KAFKA_IMAGE_VALUES;
+  do
+    name="strimzi-kafka-$(echo "$val" | cut -d'=' -f1 | tr -d '.')";
+    image=$(echo "${val}" | cut -d'=' -f2);
+    yq ea -i ".spec.relatedImages += [{\"name\": \"$name\", \"image\": \"$image\"}]" "${CSV_FILE}";
+  done
+
+  # Add auxiliary images
+  ENV="*BRIDGE* *JMXTRANS* *KANIKO_EXECUTOR*"
+  for env in $ENV;
+  do
+    image=$(yq ea ".. | select(has(\"name\")).env[] | (select (.name == \"$env\")).value" "${CSV_FILE}")
+    name="strimzi-$(echo "$env" | $SED 's/*//g' | $SED 's/_/-/g' | tr '[:upper:]' '[:lower:]')"
+    yq ea -i ".spec.relatedImages += [{\"name\": \"$name\", \"image\": \"$image\"}]" "${CSV_FILE}";
+  done
+}
+
+# Iterates through images listed in newly generated relatedImages section
+# and replaces image tags with image digests
+generate_image_digests() {
+
+  IMAGES=$(yq eval '.spec.relatedImages[].image' "${CSV_FILE}")
+  for image_tag in $IMAGES;
+  do
+    tag=$(echo "$image_tag" | cut -d':' -f2);
+    image=$(echo "$image_tag" | cut -d':' -f1);
+    repo=$(echo "$image" | rev | cut -d'/' -f1 | rev);
+    
+    if [ "$repo" != "kafka-bridge" ]; then
+      tag=$(echo "$tag" | $SED "s/$(echo "$tag" | cut -d'-' -f1)/${DOCKER_TAG}/g")
+    fi
+    registry=$DOCKER_REGISTRY;
+    org=$DOCKER_ORG;
+
+    echo "Get digest from remote registry for image: ${registry}/${org}/${repo}:${tag}"
+    resp="$(curl -s -w '\n%{http_code}' -H "Accept: application/vnd.docker.distribution.manifest.v2+json" "https://${registry}/v2/${org}/${repo}/manifests/${tag}")"
```

Comment:
When I set registry to `docker.io` (with my own repo) this line results in a `301` return code from `curl` suggesting the URL format has changed. When I use `quay.io` I get a `401` unauthorized return code. 

Is there a way to get these digests by using the `docker` or `podman` CLI's that would result in a less brittle implementation.

Reply:
For example I can run:
```
$ docker image inspect docker.io/tomncooper/operator:latest | jq -r ".[0].RepoDigests"  
[
  "tomncooper/operator@sha256:906a7e45f653957d13ff9c2727c2db98c06dab385f811ab686743fbcd1e2944a"
]
```

Reply:
This works with `podman` too. Of course the issue with this is you need to pull the images first.